### PR TITLE
hotfix(upstreams) ensure 0 is a valid value for health check fields

### PR DESCRIPTION
**Pull request by thibaultcha**
A regression introduced in 3e240b8722141d65b84ac69ab5c2bbc3522eb69d.

### CODE REVIEW

File: kong/dao/schemas/upstreams.lua [hunk 1]
```diff
@@ -94,12 +98,12 @@ local healthchecks_defaults = {
 
 local funcs = {
   timeout = check_nonnegative,
-  concurrency = check_positive_int,
+  concurrency = check_positive_int_or_zero,
```
Comment (thibaultcha) [inline kong/dao/schemas/upstreams.lua]: hold wrt to this one!

File: kong/dao/schemas/upstreams.lua [hunk 2]
```diff
@@ -14,7 +14,6 @@ local function check_nonnegative(arg)
   end
 end
 
-
```
Comment (thibaultcha) [inline kong/dao/schemas/upstreams.lua]: argh

**Comment by thibaultcha** [comment]
Manually merged, thanks!

File: kong/dao/schemas/upstreams.lua [hunk 3]
```diff
@@ -23,6 +22,20 @@ local function check_positive_int(t)
...
+local function check_positive_int_or_zero(t)
+  if t == 0 then
+    return true
+  end
+
+  local ok = check_positive_int(t)
+  if not ok then
+    return false, "must be 0 (disabled), or an integer between 1 and "
+                  .. 2 ^31 - 1
+  end
```
Comment (Tieske) [inline kong/dao/schemas/upstreams.lua]: why the "ok" here?

```lua
if not check_positive_int(t) then
  return false, "must be 0 (disabled), or an integer between 1 and "
                .. 2 ^31 - 1
end
```

**Pull request closed without merging**